src/store.ts

```diff
@@ -0,0 +1,22 @@
+import LruCache from "lru-cache";
+
+export const lru = new LruCache({ max: 1000 });
+
+// Creates a functions that memoizes its result using an LRU cache.
+//
+// The cache key is a string created from the arguments using
+// `makeKey`.
+export async function cache<V>(
+  k: string,
+  f: (key: string) => Promise<V>,
+): Promise<V> {
+  const cached = lru.get<{ value: V }>(k);
+  if (! cached) {
+    console.log(k);
```

Comment:
👀

Reply:
Still a work in progress 😄 
And had to make some changes to your implementation.